strawberryfields/engine.py

```diff
@@ -336,8 +360,144 @@ def run(self, program, *, shots=1, compile_options={}, modes=None, state_options
             # empty sequence
             pass
         else:
-            result.state = self.backend.state(modes, **state_options)  # tfbackend.state can use kwargs
+            result.state = self.backend.state(
+                modes, **state_options
+            )  # tfbackend.state can use kwargs
         return result
 
 
+class StarshipEngine(BaseEngine):
+    """
+    Starship quantum program executor engine.
+
+    Executes :class:`.Program` instances on the chosen remote backend, and makes
+    the results available via :class:`.Result`.
+
+    Args:
+        backend (str, BaseBackend): name of the backend, or a pre-constructed backend instance
+    """
+
+    API_DEFAULT_REFRESH_SECONDS = 1
```

Comment:
Should this be an `APIClient` argument?

Reply:
I think this is specific to the engine, as it sets the polling delay that the engine would wait before fetching the job status. Perhaps renaming it to `POLLING_DELAY_SECONDS` would be better?

Reply:
Make sense :+1: It was the `API` that threw me 😆 

Would this be better as a instance attribute, so that different engines can have different polling delays?

Reply:
Changed to instance attribute, and agreed that having different polling delays could come in handy.